cli/coredump/coredump.go

```diff
@@ -14,97 +17,149 @@ import (
 //go:embed scripts/*
 var corescripts embed.FS
 
-// findFile makes absolute path and checks that file exists.
-func findFile(fileName string) (string, error) {
-	filePath, err := filepath.Abs(fileName)
+//go:embed extensions
+var extensions embed.FS
+
+const packEmbedPath = "scripts/tarabrt.sh"
+const inspectEmbedPath = "scripts/gdb.sh"
+
+// Pack packs coredump into a tar.gz archive.
+func Pack(corePath string) error {
+	tmpDir, err := os.MkdirTemp(os.TempDir(), "tt-coredump-*")
 	if err != nil {
-		return "", err
-	}
-	if _, err := os.Stat(filePath); err != nil {
-		return "", err
+		return fmt.Errorf("there was some problem packing archive. "+
+			"Error: '%v'", err)
 	}
-	return filePath, err
-}
+	defer os.RemoveAll(tmpDir) // clean up on function return
 
-// Pack packs coredump into a tar.gz archive.
-func Pack(coreName string) error {
-	corePath, err := findFile(coreName)
+	scriptArgs := []string{"-c", corePath}
+
+	// Prepare gdb wrapper for packing
+	inspectPath := filepath.Join(tmpDir, filepath.Base(inspectEmbedPath))
+	err = util.FsCopyFileChangePerms(corescripts, inspectEmbedPath, inspectPath, 0755)
 	if err != nil {
 		return fmt.Errorf("there was some problem packing archive. "+
 			"Error: '%v'", err)
 	}
-	content, err := util.ReadEmbedFile(corescripts, "scripts/tarabart.sh")
+	scriptArgs = append(scriptArgs, "-g", inspectPath)
+
+	// Prepare gdb extensions for packing
+	const extDirName = "extensions"
+	extEntries, err := extensions.ReadDir(extDirName)
 	if err != nil {
 		return fmt.Errorf("there was some problem packing archive. "+
 			"Error: '%v'", err)
 	}
-	cmd := exec.Command("bash", "-s")
-	StdinPipe, err := cmd.StdinPipe()
+	for _, extEntry := range extEntries {
+		extSrc := filepath.Join(extDirName, extEntry.Name())
+		extDst := filepath.Join(tmpDir, extEntry.Name())
+		err = util.FsCopyFileChangePerms(extensions, extSrc, extDst, 0644)
+		if err != nil {
+			return fmt.Errorf("there was some problem packing archive. "+
+				"Error: '%v'", err)
+		}
+		scriptArgs = append(scriptArgs, "-x", extDst)
+	}
+
+	script, err := corescripts.Open(packEmbedPath)
 	if err != nil {
 		return fmt.Errorf("there was some problem packing archive. "+
 			"Error: '%v'", err)
 	}
-	cmd.Env = append(cmd.Env, "COREFILE_ENV="+corePath)
+	cmdArgs := []string{"-s", "--"}
+	cmd := exec.Command("bash", append(cmdArgs, scriptArgs...)...)
+	cmd.Stdin = script
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	err = cmd.Start()
-	StdinPipe.Write([]byte(content))
-	StdinPipe.Close()
+	if err != nil {
+		return fmt.Errorf("there was some problem packing archive. "+
+			"Error: '%v'", err)
+	}
 	err = cmd.Wait()
-	if err == nil {
-		log.Infof("Core was successfully packed.")
-	} else {
-		err = fmt.Errorf("there was some problem packing archive. "+
+	if err != nil {
```

Comment:
Let's use `Run`, which is Start() + Wait() in one call.

Reply:
Done